kernel/memory/src/paging/mod.rs

```diff
@@ -254,6 +256,22 @@ pub fn init(
     let mut higher_half_mapped_pages: [Option<NoDrop<MappedPages>>; 32] = Default::default();
     let mut identity_mapped_pages:    [Option<NoDrop<MappedPages>>; 32] = Default::default();
 
+    let stack_end_virt = stack_start_virt + boot_info.stack_size()?;
+    // Stack frames are not guaranteed to be contiguous.
+    let mut stack_mappings = [None; 34];
+    // + 1 accounts for the guard page which does not have a corresponding frame.
+    for (i, page) in ((Page::containing_address(stack_start_virt) + 1)..=Page::containing_address(stack_end_virt - 1)).enumerate() {
```

Comment:
for convenience, you can use [`PageRange::from_virt_addr()`](https://www.theseus-os.com/Theseus/doc/memory_structs/struct.PageRange.html#method.from_virt_addr)